components/iconSelector.js

```diff
@@ -25,108 +25,221 @@ const StyledPressable = styled(Animated.createAnimatedComponent(Pressable));
+        for (let i = 0; i < numRows; i++) {
+            if (icons[i]) {
+                items.push(
+                    <StyledOpacity
+                        key={`icon${i}`}
+                        onPress={() => {
+                            setIcon(icons[i]);
+                            toggleSelector(false);
+                            close();
+                        }}
+                        className="w-[60px] h-[60px] items-center justify-center mb-1"
+                    >
+                        <StyledIcon
+                            key={i}
+                            name={icons[i]}
+                            size={40}
+                            color={'#ffffff'}
+                        />
+                    </StyledOpacity>
+                );
+            } else {
+                items.push(
+                    <StyledView
+                        key={`icon${i}`}
+                        className="w-[60px] h-[60px] mb-1"
+                    ></StyledView>
+                );
+            }
+        }
+        return items;
+    }
 
-	return (
-		<>
-			<StyledPressable
-				style={{ opacity: bgOpacityInterpolation }}
-				pointerEvents={opened ? 'auto' : 'none'}
-				onPress={() => {
-					toggleSelector(false);
-					close();
-				}}
-				className='absolute top-0 left-0 bg-offblack w-screen h-screen'
-			/>
-			<StyledAnimView
-				style={{ opacity: opacityInterpolation }}
-				pointerEvents={opened ? 'auto' : 'none'}
-				className='absolute -translate-x-[150px] left-1/2 top-[60%] -translate-y-[250px] w-[80%] p-[15px] max-w-[300px] h-[50%] max-h-[500px] bg-offblack border border-[#3D3D3D] rounded-[20px]'
-			>
-				<StyledText className='text-offwhite font-bold text-3xl text-center mb-3'>
-					Select an Icon
-				</StyledText>
-				<StyledScrollView>
-					<StyledView className='flex justify-around flex-row flex-wrap'>
-						{iconGallery(icons)}
-					</StyledView>
-				</StyledScrollView>
-			</StyledAnimView>
-		</>
-	);
+    return (
+        <>
+            <StyledPressable
+                style={{ opacity: bgOpacityInterpolation }}
+                pointerEvents={opened ? 'auto' : 'none'}
+                onPress={() => {
+                    toggleSelector(false);
+                    close();
+                }}
+                className="absolute top-0 left-0 bg-offblack w-screen h-screen"
+            />
+            <StyledAnimView
+                style={{ opacity: opacityInterpolation }}
+                pointerEvents={opened ? 'auto' : 'none'}
+                className="absolute -translate-x-[150px] left-1/2 top-[60%] -translate-y-[280px] w-[80%] p-[15px] max-w-[300px] h-[85%] max-h-[500px] bg-offblack border border-[#3D3D3D] rounded-[20px]"
+            >
+                <StyledText className="text-offwhite font-bold text-3xl text-center mb-3">
+                    Select an Icon
+                </StyledText>
+                <StyledScrollView>
```

Comment:
The load times are really slow. This scroll view should be changed out with a FlatList

Reply:
Mine load immediately, I will test changing